### PR TITLE
Minor stylistic changes

### PR DESCRIPTION
**Pull request by bishtgautam**
- Adds `PetscCall()`
- Defines local variables closer to the code where they are used.

### CODE REVIEW

File: swe_roe/ex1.c [hunk 1]
```diff
@@ -183,141 +169,137 @@ PetscErrorCode RHSFunction(TS ts, PetscReal t, Vec X, Vec F, void *ptr) {
...
+  for (PetscInt j = user->gys + 1; j < user->gys + user->gym; j = j + 1) {
+    for (PetscInt i = user->gxs + 1; i < user->gxs + user->gxm; i = i + 1) {
       /* - - - - - - - - - - - - - - - *
        * Compute fluxes in x-driection *
        * - - - - - - - - - - - - - - - */
+      PetscScalar fij[3];
 
-      sn = 0.0;
-      cn = 1.0;
+      PetscReal sn = 0.0;
```
Comment (donghuix) [inline swe_roe/ex1.c:245]: Do we want to declare the variables in the for loop?

**Review by donghuix — COMMENTED** [review]
Many variables are declared in the for loop, is this OK?

File: swe_roe/ex1.c (hunk 1 above)
Comment (bishtgautam) [reply swe_roe/ex1.c:245]: Yes, it is okay to define variables in a loop. We will eventually come up with a style guide for the project.

Comment (donghuix) [reply swe_roe/ex1.c:245]: Ok, sounds good to me.

**Review by donghuix — APPROVED** [review]

**Pull request merged**